web/jobs/jobs.go

```diff
@@ -259,7 +273,38 @@ func contactSupport(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
-func newTrigger(c echo.Context) error {
+func (h *HTTPHandler) sendMarketingEmail(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+
+	if err := middlewares.Allow(c, permission.POST, &job.JobRequest{WorkerType: "sendmail"}); err != nil {
+		return err
+	}
+
+	req := apiMarketing{}
+	if _, err := jsonapi.Bind(c.Request().Body, &req); err != nil {
+		return wrapJobsError(err)
+	}
+	subject, _ := req.Arguments["subject"].(string)
+	if subject == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "Invalid subject")
+	}
+	parts, _ := req.Arguments["parts"].([]mail.Part)
```

Comment:
It won't work. Currently, `req.Arguments` is defined as `map[string]interface{}`. When Go unmarshals JSON to it, it will only uses the basic types (nil, bool, float64, string, slices and maps). I think you can use a struct for `req.Arguments` to fix this issue.